shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,45 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.DTO.ProductCategoryDTO;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.List;
+
+@RestController
+public class ProductController {
+    ProductService productService;
```

Comment:
Here you don't seem to have injected the service. Use the lombok adnotation together with the final keyword